### ISSUE TITLE
Gun spawns with wrong ammo loaded

### ISSUE THREAD

**Issue by Yinyang107** [CleverRaven/Cataclysm-DDA#55709]
### Describe the bug

A Glock 22 spawned with an attached Glock .40 S&W/.357 SIG 22-round magazine. The magazine was loaded with .357 SIG ammo, with which the magazine is compatible, but the gun is not.

### Steps to reproduce

1. Loot a zombie and find a Glock 22
2. Magazine may be full of the wrong ammo
3. (Optional) forget to make sure your loaded gun is loaded with the right bullets and die to a Mi-go

### Expected behavior

I expected that bullets that spawned pre-loaded into the gun would be compatible with that gun.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19043.1526 (21H1)
- Game Version: 582984d [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    Magiclysm [magiclysm],
    Mythical Martial Arts [MMA],
    sees-player, run, crouch, bleed icons, retrodays [sees_player_retro],
    No Fungal Growth [no_fungal_growth],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    My Sweet Cataclysm [my_sweet_cataclysm],
    No Rail Stations [No_Rail_Stations],
    Faster Crop Growth [Yin_Faster_crops]
]

**Comment by LeahLuong** [comment]
I don't believe the game lets you put a clip loaded w/ the wrong bullets into a gun. Something inimitably chthonic must have been done to that pistol!

**Comment by KurzedMetal** [comment]
I never noticed something like this myself playing unmodded, maybe it's related to one of his mods?
